IonDotnet/Internals/Text/TextScanner.cs

```diff
@@ -998,6 +998,8 @@ public int LoadDoubleQuotedString(StringBuilder sb, bool isClob)
                     case CharacterSequence.CharSeqEscapedNewlineSequence3:
                         continue;
                     case -1:
+                        FinishNextToken(isClob ? Token : TextConstants.TokenStringDoubleQuote, isClob);
```

Comment:
Why is this case different for clobs? Isn't it true that if EOF is reached before the closing `"` in either case, an error should be raised?

Reply:
I tried to keep the previous logic here, as before for ``-1`` and ``"`` we had the same actions.

The difference is that ``FinishNextToken`` is to mark what is the current token and whether it is finished or not. 
If we have a string inside a clob, the token is "clob" (as we are still parsing the contents of the clob), but if it is not a clob, then the token is just a " 

I double checked and in case of a clobs, such cases are handled in #9 (Fix clobs issues)